File: AddItem.cs
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using MySql.Data.MySqlClient;

namespace MilkTeaOrderAndInventorySystem
{

    public partial class AddItem
    {
        private int flag = 0;

        public string passItemCategory { get; set; }
        public bool checker { get; set; } = false;
        public string itemID { get; set; }

        private readonly OpenFileDialog opf = new OpenFileDialog() { Filter = "Choose Image(*.JPG;*.PNG;*.GIF)|*.jpg;*.png;*.gif" };

        private const int CS_DROPSHADOW = 131072;

        protected override CreateParams CreateParams
        {
            get
            {
                var cp = base.CreateParams;
                cp.ClassStyle = cp.ClassStyle | CS_DROPSHADOW;
                return cp;
            }
        }

        public AddItem()
        {
            InitializeComponent();
        }

        private void ManageItem_Load(object sender, EventArgs e)
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }
            UniversalVariables.conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb";

            if (checker == true)
            {
                setFields();
                lblItems.Text = "Edit Product";
            }
        }

        private bool isProductExist(string item)
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            try
            {
                if (checker == true)
                {
                    UniversalVariables.sql = "SELECT * FROM storeItems WHERE prodName = '" + item + "' AND sID <> '" + itemID + "'";
                    UniversalVariables.connect();
                }
                else
                {
                    UniversalVariables.sql = "SELECT * FROM storeItems WHERE prodName = '" + item + "'";
                    UniversalVariables.connect();
                }

                if (UniversalVariables.dr.Read())
                {
                    lblChecker.Visible = true;
                    lblChecker.Text = "An item with this name already exist!";

                    return true;
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }
            UniversalVariables.conn.Close();

            return false;
        }

        private void setFields()
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            try
            {
                UniversalVariables.sql = "SELECT * FROM storeItems WHERE sID = '" + itemID + "'";
                UniversalVariables.connect();

                if (UniversalVariables.dr.Read())
                {
                    byte[] arrimage = (byte[])UniversalVariables.dr["prodPic"];
                    var mstream = new System.IO.MemoryStream(arrimage);
                    itemPicture.Image = Image.FromStream(mstream);

                    txtItemName.Text = UniversalVariables.dr.GetString("prodName").ToString();
                    txtItemPrice.Text = UniversalVariables.dr.GetString("prodPrice").ToString();
                    txtItemQuantity.Text = UniversalVariables.dr.GetString("prodAvail").ToString();
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }
            UniversalVariables.conn.Close();
        }

        private void updateItem()
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            try
            {
                byte[] arrimage1;
                var originalImage = itemPicture.Image;
                var mstream1 = new System.IO.MemoryStream();
                originalImage.Save(mstream1, System.Drawing.Imaging.ImageFormat.Jpeg);
                arrimage1 = mstream1.GetBuffer();
                mstream1.Close();

                string itemCat = passItemCategory;
                string itemName = txtItemName.Text;
                string itemPrice = txtItemPrice.Text;
                string itemCount = txtItemQuantity.Text;

                UniversalVariables.sql = "UPDATE storeItems SET prodPic = @prodPic, prodName = @prodName, prodPrice = @prodPrice, prodAvail = @prodAvail WHERE sID = '" + itemID + "'";
                {
                    ref var withBlock = ref UniversalVariables.cmd;
                    withBlock.Parameters.Clear();
                    withBlock.Parameters.Add("@prodPic", MySqlDbType.Blob).Value = arrimage1;
                    withBlock.Parameters.Add("@prodCat", MySqlDbType.VarChar).Value = itemCat;
                    withBlock.Parameters.Add("@prodName", MySqlDbType.VarChar).Value = itemName;
                    withBlock.Parameters.Add("@prodPrice", MySqlDbType.VarChar).Value = itemPrice;
                    withBlock.Parameters.Add("@prodAvail", MySqlDbType.VarChar).Value = itemCount;
                }
                UniversalVariables.connect();

                UniversalVariables.dr.Close();

                if (Conversions.ToBoolean(UniversalVariables.cmd.ExecuteNonQuery()))
                {
                    Close();
                    var obj1 = new SavedConfirmation();
                    obj1.ItemAddedConfirmation = "Item Updated Succesfully!";
                    obj1.ShowDialog();
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }
            UniversalVariables.conn.Close();
        }

        private void insertNewItem()
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            try
            {
                byte[] arrimage1;
                var originalImage = itemPicture.Image;
                var mstream1 = new System.IO.MemoryStream();
                originalImage.Save(mstream1, System.Drawing.Imaging.ImageFormat.Jpeg);
                arrimage1 = mstream1.GetBuffer();
                mstream1.Close();

                string itemCat = passItemCategory;
                string itemName = txtItemName.Text;
                string itemPrice = txtItemPrice.Text;
                string itemCount = txtItemQuantity.Text;

                UniversalVariables.sql = "INSERT INTO storeItems (prodPic, prodCat, prodName, prodPrice, prodAvail) VALUES (@prodPic, @prodCat, @prodName, @prodPrice, @prodAvail)";
                {
                    ref var withBlock = ref UniversalVariables.cmd;
                    withBlock.Parameters.Clear();
                    withBlock.Parameters.Add("@prodPic", MySqlDbType.Blob).Value = arrimage1;
                    withBlock.Parameters.Add("@prodCat", MySqlDbType.VarChar).Value = itemCat;
                    withBlock.Parameters.Add("@prodName", MySqlDbType.VarChar).Value = itemName;
                    withBlock.Parameters.Add("@prodPrice", MySqlDbType.VarChar).Value = itemPrice;
                    withBlock.Parameters.Add("@prodAvail", MySqlDbType.VarChar).Value = itemCount;
                }
                UniversalVariables.connect();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }
            UniversalVariables.conn.Close();
        }

        private void setItemPic()
        {
            if (opf.ShowDialog() == DialogResult.OK)
            {
                flag = 1;
                var originalImage = Image.FromFile(opf.FileName);
                itemPicture.Image = originalImage;
            }
        }

        private void btnDiscard_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            if (checker == true)
            {
                flag = 1;
            }

            if (string.IsNullOrEmpty(txtItemName.Text) | string.IsNullOrEmpty(txtItemPrice.Text) | string.IsNullOrEmpty(txtItemQuantity.Text) | flag == 0)
            {
                lblChecker.Visible = true;
            }
            else if (!isProductExist(txtItemName.Text))
            {
                if (checker == false)
                {
                    insertNewItem();
                    flag = 0;
                }
                else
                {
                    updateItem();
                    checker = false;
                }
                DialogResult = DialogResult.OK;
            }
        }

        private void setPic_Click(object sender, EventArgs e)
        {
            setItemPic();
            lblChecker.Visible = false;
        }

        private void txtItemName_TextChanged(object sender, EventArgs e)
        {
            lblChecker.Visible = false;
        }

        private void txtItemPrice_TextChanged(object sender, EventArgs e)
        {
            lblChecker.Visible = false;
        }

        private void txtItemQuantity_TextChanged(object sender, EventArgs e)
        {
            lblChecker.Visible = false;
        }

        private void txtItemPrice_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsNumber(e.KeyChar) && !char.IsControl(e.KeyChar))
                e.KeyChar = Conversions.ToChar("");
        }

        private void txtItemQuantity_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsNumber(e.KeyChar) && !char.IsControl(e.KeyChar))
                e.KeyChar = Conversions.ToChar("");
        }
    }
}

File: AddItem.vb
Imports MySql.Data.MySqlClient

Public Class AddItem
    Dim flag As Integer = 0

    Public Property passItemCategory As String
    Public Property checker As Boolean = False
    Public Property itemID As String

    ReadOnly opf As New OpenFileDialog With {
        .Filter = "Choose Image(*.JPG;*.PNG;*.GIF)|*.jpg;*.png;*.gif"
    }

    Private Const CS_DROPSHADOW As Integer = 131072

    Protected Overrides ReadOnly Property CreateParams() As System.Windows.Forms.CreateParams
        Get
            Dim cp As CreateParams = MyBase.CreateParams
            cp.ClassStyle = cp.ClassStyle Or CS_DROPSHADOW
            Return cp
        End Get
    End Property

    Private Sub ManageItem_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If
        conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb"

        If checker = True Then
            setFields()
            lblItems.Text = "Edit Product"
        End If
    End Sub

    Private Function isProductExist(ByVal item As String) As Boolean
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Try
            If checker = True Then
                sql = "SELECT * FROM storeItems WHERE prodName = '" & item & "' AND sID <> '" & itemID & "'"
                connect()
            Else
                sql = "SELECT * FROM storeItems WHERE prodName = '" & item & "'"
                connect()
            End If

            If dr.Read Then
                lblChecker.Visible = True
                lblChecker.Text = "An item with this name already exist!"

                Return True
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        conn.Close()

        Return False
    End Function

    Private Sub setFields()
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Try
            sql = "SELECT * FROM storeItems WHERE sID = '" & itemID & "'"
            connect()

            If dr.Read Then
                Dim arrimage() As Byte = dr.Item("prodPic")
                Dim mstream As New System.IO.MemoryStream(arrimage)
                itemPicture.Image = System.Drawing.Image.FromStream(mstream)

                txtItemName.Text = dr.GetString("prodName").ToString
                txtItemPrice.Text = dr.GetString("prodPrice").ToString
                txtItemQuantity.Text = dr.GetString("prodAvail").ToString
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        conn.Close()
    End Sub

    Private Sub updateItem()
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Try
            Dim arrimage1() As Byte
            Dim originalImage = itemPicture.Image
            Dim mstream1 As New System.IO.MemoryStream
            originalImage.Save(mstream1, System.Drawing.Imaging.ImageFormat.Jpeg)
            arrimage1 = mstream1.GetBuffer
            mstream1.Close()

            Dim itemCat As String = passItemCategory
            Dim itemName As String = txtItemName.Text
            Dim itemPrice As String = txtItemPrice.Text
            Dim itemCount As String = txtItemQuantity.Text

            sql = "UPDATE storeItems SET prodPic = @prodPic, prodName = @prodName, prodPrice = @prodPrice, prodAvail = @prodAvail WHERE sID = '" & itemID & "'"
            With cmd
                .Parameters.Clear()
                .Parameters.Add("@prodPic", MySqlDbType.Blob).Value = arrimage1
                .Parameters.Add("@prodCat", MySqlDbType.VarChar).Value = itemCat
                .Parameters.Add("@prodName", MySqlDbType.VarChar).Value = itemName
                .Parameters.Add("@prodPrice", MySqlDbType.VarChar).Value = itemPrice
                .Parameters.Add("@prodAvail", MySqlDbType.VarChar).Value = itemCount
            End With
            connect()

            dr.Close()

            If cmd.ExecuteNonQuery Then
                Me.Close()
                Dim obj1 As New SavedConfirmation
                obj1.ItemAddedConfirmation = "Item Updated Succesfully!"
                obj1.ShowDialog()
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        conn.Close()
    End Sub

    Private Sub insertNewItem()
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Try
            Dim arrimage1() As Byte
            Dim originalImage = itemPicture.Image
            Dim mstream1 As New System.IO.MemoryStream
            originalImage.Save(mstream1, System.Drawing.Imaging.ImageFormat.Jpeg)
            arrimage1 = mstream1.GetBuffer
            mstream1.Close()

            Dim itemCat As String = passItemCategory
            Dim itemName As String = txtItemName.Text
            Dim itemPrice As String = txtItemPrice.Text
            Dim itemCount As String = txtItemQuantity.Text

            sql = "INSERT INTO storeItems (prodPic, prodCat, prodName, prodPrice, prodAvail) VALUES (@prodPic, @prodCat, @prodName, @prodPrice, @prodAvail)"
            With cmd
                .Parameters.Clear()
                .Parameters.Add("@prodPic", MySqlDbType.Blob).Value = arrimage1
                .Parameters.Add("@prodCat", MySqlDbType.VarChar).Value = itemCat
                .Parameters.Add("@prodName", MySqlDbType.VarChar).Value = itemName
                .Parameters.Add("@prodPrice", MySqlDbType.VarChar).Value = itemPrice
                .Parameters.Add("@prodAvail", MySqlDbType.VarChar).Value = itemCount
            End With
            connect()
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        conn.Close()
    End Sub

    Private Sub setItemPic()
        If opf.ShowDialog = Windows.Forms.DialogResult.OK Then
            flag = 1
            Dim originalImage = Image.FromFile(opf.FileName)
            itemPicture.Image = originalImage
        End If
    End Sub

    Private Sub btnDiscard_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        Me.DialogResult = DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        If checker = True Then
            flag = 1
        End If

        If txtItemName.Text = Nothing Or txtItemPrice.Text = Nothing Or txtItemQuantity.Text = Nothing Or flag = 0 Then
            lblChecker.Visible = True
        Else
            If Not isProductExist(txtItemName.Text) Then
                If checker = False Then
                    insertNewItem()
                    flag = 0
                Else
                    updateItem()
                    checker = False
                End If
                Me.DialogResult = DialogResult.OK
            End If
        End If
    End Sub

    Private Sub setPic_Click(sender As Object, e As EventArgs) Handles itemPicture.Click
        setItemPic()
        lblChecker.Visible = False
    End Sub

    Private Sub txtItemName_TextChanged(sender As Object, e As EventArgs) Handles txtItemName.TextChanged
        lblChecker.Visible = False
    End Sub

    Private Sub txtItemPrice_TextChanged(sender As Object, e As EventArgs) Handles txtItemPrice.TextChanged
        lblChecker.Visible = False
    End Sub

    Private Sub txtItemQuantity_TextChanged(sender As Object, e As EventArgs) Handles txtItemQuantity.TextChanged
        lblChecker.Visible = False
    End Sub

    Private Sub txtItemPrice_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtItemPrice.KeyPress
        If Not Char.IsNumber(e.KeyChar) AndAlso Not Char.IsControl(e.KeyChar) Then e.KeyChar = ""
    End Sub

    Private Sub txtItemQuantity_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtItemQuantity.KeyPress
        If Not Char.IsNumber(e.KeyChar) AndAlso Not Char.IsControl(e.KeyChar) Then e.KeyChar = ""
    End Sub
End Class